### PR TITLE
[gym] Add support for exporting pkg file.

### PR DESCRIPTION
**Pull request by yokoe**
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:
- [ ] Run `bundle exec rspec` from the subdirectory of each tool you modified. Alternatively, run `rake test_all` from the root directory.
- [ ] Run `bundle exec rubocop -a` to ensure the code style is valid
- [ ] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:

---

Related to #6673. This PR adds missing steps for generating pkg file in macOS project to gym. The steps are almost same to that for iOS project.

We can generate pkg file from xcarchive in the same way exporting ipa file in iOS project using xcodebuild with "--exportArchive" option.

### CODE REVIEW

**Review by ohayon — DISMISSED** [review]
Thanks for the submission @yokoe!

File: gym/lib/gym/runner.rb [hunk 1]
```diff
@@ -28,6 +28,8 @@ def run
 
         path
       elsif Gym.project.mac?
+        package_app
```
Comment (ohwutup) [inline gym/lib/gym/runner.rb]: I don't think that we want to do this here for every app. There are times where this is not going to be desired behavior. Instead, I think there should be some argument passed to _gym_ to make this happen.

**Review by ohayon — CHANGES_REQUESTED** [review]

**Comment by ohwutup** [comment]
Thanks for the submission @yokoe, I think the approach is a bit too heavy handed to generate the `pkg` for every project built for macOS. Let me know what you think! 🐙

**Comment by yokoe** [comment]
Thanks for the review. 

I think the steps should be linked to the `export_method` option. I've updated the source code, and now the steps never run unless `export_method` was specified explicitly.
I suppose `export_method` option is not specified in most macOS app projects because it has no effect currently, and without specifying it, gym will run in the same way, so I think this change will not do much harm.

I also considered adding a new option but it seems redundant.

File: gym/lib/gym/runner.rb [hunk 2]
```diff
@@ -28,8 +28,10 @@ def run
 
         path
       elsif Gym.project.mac?
-        package_app
-        move_pkg
+        if Gym.config[:export_method]
```
Comment (ohwutup) [inline gym/lib/gym/runner.rb]: should we at least check that the export method is of a certain kind?

**Comment by yokoe** [comment]
I modified the source code and now packaging steps run only if export_method is 'app-store' or 'developer-id'.

Thanks to @triplef
https://github.com/fastlane/fastlane/issues/6673#issuecomment-256959890

**Comment by ohwutup** [comment]
Hey @yokoe, apologies for the delay here... I think I am missing a bit of knowledge about this situation: Is there a case in which someone would want to generate both a `.pkg` and a `.app` in the same execution of _gym_? 

Also, regarding the `export_method` option, I think this is using the option a little bit differently than how it is used for iOS builds as this option implies the configuration in which the app should be built (i.e. Debug or Release).

**Comment by ohwutup** [comment]
Hey @yokoe, I am going to close this PR as there hasn't been activity for a while and we have changed around some things now that we are only publishing one gem, _fastlane_. Feel free to make some changes as mentioned above and fix the merge conflicts and we can reopen this or you can open a new PR.

Thanks again for your contribution! 🚀

**Pull request closed without merging**